File: tools/qvod/src/test/java/top/bettercode/summer/tools/qvod/test/QTestController.java
package top.bettercode.summer.tools.qvod.test;


import org.springframework.validation.annotation.Validated;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RestController;
import top.bettercode.summer.web.BaseController;
import top.bettercode.summer.web.form.FormDuplicateCheck;

/**
 * @author Peter Wu
 */
@RestController
@Validated
public class QTestController extends BaseController {

  @FormDuplicateCheck
  @RequestMapping(value = "/test")
  public Object test() {
    return ok(new DataBean());
  }

}

File: tools/qvod/src/test/kotlin/top/bettercode/summer/tools/qvod/QTestController.kt
package top.bettercode.summer.tools.qvod

import org.springframework.validation.annotation.Validated
import org.springframework.web.bind.annotation.RequestMapping
import org.springframework.web.bind.annotation.RestController
import top.bettercode.summer.web.BaseController
import top.bettercode.summer.web.form.FormDuplicateCheck

/**
 * @author Peter Wu
 */
@RestController
@Validated
class QTestController : BaseController() {
    @FormDuplicateCheck
    @RequestMapping(value = ["/test"])
    fun test(): Any {
        return ok(DataBean())
    }
}